dynamiq/nodes/agents/react.py

```diff
@@ -543,13 +543,21 @@ def _extract_final_answer_xml(self, llm_output: str) -> str:
         return final_answer["answer"]
 
     def generate_input_formats(self, tools: list[Node]) -> str:
+        """Generate formatted input descriptions for each tool."""
         input_formats = []
         for tool in tools:
-            params = [
-                f"{name} ({field.annotation.__name__}): {field.description or 'No description'}"
-                for name, field in tool.input_schema.model_fields.items()
-                if not field.json_schema_extra or field.json_schema_extra.get("is_accessible_to_agent", True)
-            ]
+            params = []
+            for name, field in tool.input_schema.model_fields.items():
+                if not field.json_schema_extra or field.json_schema_extra.get("is_accessible_to_agent", True):
+                    # Handle Union types
+                    if hasattr(field.annotation, "__origin__") and field.annotation.__origin__ is Union:
+                        type_names = [t.__name__ for t in field.annotation.__args__ if hasattr(t, "__name__")]
+                        type_str = " | ".join(type_names)
```

Comment:
This is a little bit clear. Plase check if works fine for you.
Don't forget to make proper imports:
```
import types
from typing import get_origin
```
```suggestion
                    if get_origin(field.annotation) in (Union, types.UnionType):
                        type_str = str(field.annotation)
```